libs/fundamental-redirects/index.js

```diff
@@ -1147,6 +1147,37 @@ for (const [pattern, path] of [
   );
 }
 
+function simpleRedirect(
+  pattern,
+  replacement,
+  { permanent = true, colonToSlash = false } = {}
+) {
+  return (path) => {
+    const match = pattern.exec(path);
+    if (match === null) {
+      return null;
+    }
+    const status = permanent ? 301 : 302;
+    let url = path.replace(pattern, replacement);
+    if (colonToSlash) {
+      url = url.replace(/:/g, "/");
+    }
+    return { url, status };
+  };
+}
+
+const SIMPLE_REDIRECT_PATTERNS = [
+  simpleRedirect(
+    /\/docs\/Core_JavaScript_1.5_?/,
+    "/docs/Web/JavaScript/",
+    // This will convert :
+    //   /en-US/docs/Core_JavaScript_1.5_Reference:Statements:block
+    // to:
+    //   /en-US/docs/Core_JavaScript_1.5_Reference/Statements/block
```

Comment:
This would override:
/en-US/docs/Core_JavaScript_1.5_Guide/Core_Language_Features -> /en-US/docs/Web/JavaScript/Guide/Core_Language_Features

which should be
/en-US/docs/Core_JavaScript_1.5_Guide/Core_Language_Features -> /en-US/docs/Web/JavaScript/Guide/Grammar_and_types

Actually I see that we should fix that `resolve` behaves different during build than in prod. In prod the fundamental comes first, during build the `_redirects.txt`.

Reply:
That's right! There are lots of entries in `_redirects.txt` where the conversion isn't as simple as replacing `docs/Core_JavaScript_1.5_Reference` for `docs/Web/JavaScript/Reference`. 
I.e. the pattern isn't all that simple. But I suspect it's because of other optimizations we've done in our redirects. I.e. making all to-URLs be the final URL. 

But my thinking was that that's OK. There are relatively few of these links anyway. In other words, under the circumstances I'm ready to "cheat" a bit.

Reply:
https://github.com/mdn/yari/pull/2709 this should help. And we need to add the _new_ redirects, like 
```
/en-US/docs/Web/JavaScript/Guide/Core_Language_Features -> /en-US/docs/Web/JavaScript/Guide/Grammar_and_types

Reply:
I don't understand. What does that redirect pair example have to do with `/en-US/docs/Core_JavaScript_1.5_Guide/`?